lib/primitives/address.js

```diff
@@ -277,8 +310,11 @@ class Address {
    */
 
   toString(network) {
-    if (this.version !== -1)
-      return this.toBech32(network);
+    if (this.version !== -1) {
+      if(this.version === 0)
+        return this.toBech32(network);
+      return this.toBech32m(network);
+    }
```

Comment:
nits: spacing for readability 
```suggestion
    if (this.version !== -1) {
      if (this.version === 0)
        return this.toBech32(network);

      return this.toBech32m(network);
    }
```